templates/index.test.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChildProcess, spawn, SpawnOptions } from 'child_process';
+import waitOn from 'wait-on';
+
+import Ledger from '@daml/ledger';
+import { User } from '@daml.js/create-daml-app';
+import { computeCredentials } from './Credentials';
+
+import puppeteer, { Browser, Page } from 'puppeteer';
+
+const SANDBOX_PORT = 6865;
```

Comment:
Hardcoded port numbers make me sad. You’ll have to make the tests exclusive if you want to merge this but ideally just stop hardcoding them and use `--port-file`.

Reply:
What do you mean by making them exclusive? I just copied this file unchanged, but agree we should not depend on specific port numbers.

Reply:
If you add `tags = ["exclusive"]` to a Bazel test it will not run in parallel with other tests. Running two tests with hardcoded port numbers in parallel is obviously not going to work if those port numbers collide so that’s one option to workaround it.

Reply:
I see, thanks. One option is to separate the GSG test and make it exclusive at first, and then I'll make the port file change in a follow up PR.

Reply:
I've looked into this and it shouldn't be too hard. There are just a couple of hardcoded JSON API port numbers in `ui/package.json` that I need to figure out.